estimagic/tests/optimization/test_pounders.py

```diff
@@ -3,72 +3,88 @@
 import sys
 
 import numpy as np
+import pandas as pd
 import pytest
 
-from estimagic.optimization.pounders import minimize_pounders
+from estimagic.optimization.pounders import minimize_pounders_np
 
 pytestmark = pytest.mark.skipif(
     sys.platform == "win32", reason="Pounders is not supported on Windows."
 )
 
 
+NUM_AGENTS = 10_000
+
+
+def get_random_params(length, low=0, high=1, lower_bound=-np.inf, upper_bound=np.inf):
+    params = pd.DataFrame(
+        {
+            "value": np.random.uniform(low, high, size=length),
+            "lower": lower_bound,
+            "upper": upper_bound,
+            "_internal_free": True,
+        }
+    )
+
+    return params
+
+
 def test_robustness_1():
     np.random.seed(5470)
-    true_paras = np.random.uniform(size=3)
-    start = np.random.uniform(size=3)
-    num_agents = 10000
+    true_params = get_random_params(3)
+    start_params = get_random_params(3)
+    bounds = tuple(true_params[["lower", "upper"]].to_numpy().T)
 
-    exog, endog = _simulate_sample(num_agents, true_paras)
+    exog, endog = _simulate_sample(NUM_AGENTS, true_params)
     objective = functools.partial(_nonlinear_criterion, endog, exog)
-    len_out = len(objective(start))
-    minimize_pounders(objective, start, len_out)
+    minimize_pounders_np(objective, start_params["value"].to_numpy(), bounds)
```

Comment:
Originally, this test did not have an assert because the precision was not good enough. Now that the bug is fixed, we could maybe introduce one. At least if we keep the large sample size.